src/flyte/_internal/imagebuild/docker_builder.py

```diff
@@ -46,44 +47,76 @@
 
 UV_LOCK_WITHOUT_PROJECT_INSTALL_TEMPLATE = Template("""\
 RUN --mount=type=cache,sharing=locked,mode=0777,target=/root/.cache/uv,id=uv \
-    --mount=type=bind,target=uv.lock,src=$UV_LOCK_PATH \
-    --mount=type=bind,target=pyproject.toml,src=$PYPROJECT_PATH \
-    $SECRET_MOUNT \
-    uv sync --active $PIP_INSTALL_ARGS
+   --mount=type=bind,target=uv.lock,src=$UV_LOCK_PATH \
+   --mount=type=bind,target=pyproject.toml,src=$PYPROJECT_PATH \
+   $SECRET_MOUNT \
+   uv sync --active $PIP_INSTALL_ARGS
 """)
 
 UV_LOCK_INSTALL_TEMPLATE = Template("""\
 COPY $PYPROJECT_PATH $PYPROJECT_PATH
 RUN --mount=type=cache,sharing=locked,mode=0777,target=/root/.cache/uv,id=uv \
-    $SECRET_MOUNT \
-    uv sync --active $PIP_INSTALL_ARGS --project $PYPROJECT_PATH
+   $SECRET_MOUNT \
+   uv sync --active $PIP_INSTALL_ARGS --project $PYPROJECT_PATH
+""")
+
+POETRY_lOCK_WITHOUT_PROJECT_INSTALL_TEMPLATE = Template("""\
+RUN --mount=type=cache,sharing=locked,mode=0777,target=/root/.cache/uv,id=uv \
+   $SECRET_MOUNT \
+   uv pip install poetry
+
+ENV POETRY_CACHE_DIR=/tmp/poetry_cache \
+   POETRY_VIRTUALENVS_IN_PROJECT=true
+
+RUN --mount=type=cache,sharing=locked,mode=0777,target=/tmp/poetry_cache,id=poetry \
+   --mount=type=bind,target=poetry.lock,src=$POETRY_LOCK_PATH \
+   --mount=type=bind,target=pyproject.toml,src=$PYPROJECT_PATH \
+   $SECRET_MOUNT \
+   poetry install $POETRY_INSTALL_ARGS
+""")
+
+POETRY_LOCK_INSTALL_TEMPLATE = Template("""\
+RUN --mount=type=cache,sharing=locked,mode=0777,target=/root/.cache/uv,id=uv \
+   $SECRET_MOUNT \
+   uv pip install poetry
+
+COPY $PYPROJECT_PATH $PYPROJECT_PATH
```

Comment:
I don't think this is needed right?  @pingsutw right?   This is the install-template.  it shouldn't need to copy... it should install.

Reply:
not sure, i expected the image has the code as well